scripts/get-transactions.js

```diff
@@ -1,285 +1,376 @@
-  DXdaoSnapshot.reputation.internalTxs = DXdaoSnapshot.reputation.internalTxs.concat(transactionsFetched.internalTxs)
-  
-  console.log('Getting txs from genesisProtocol..')
-  transactionsFetched = await getTransactions(genesisProtocol.address, fromBlock, toBlock);
-  DXdaoSnapshot.genesisProtocol.txs = DXdaoSnapshot.genesisProtocol.txs.concat(transactionsFetched.txs)
-  DXdaoSnapshot.genesisProtocol.internalTxs = DXdaoSnapshot.genesisProtocol.internalTxs.concat(transactionsFetched.internalTxs)
-  
-  console.log('Getting events info for controller')
-  DXdaoSnapshot.controller.events = DXdaoSnapshot.controller.events.concat( await dxController.getPastEvents(
-    'allEvents', {fromBlock: fromBlock, toBlock: toBlock}
-  ));
-  console.log('Getting events info for avatar')
-  DXdaoSnapshot.avatar.events = DXdaoSnapshot.avatar.events.concat( await dxAvatar.getPastEvents(
-    'allEvents', {fromBlock: fromBlock, toBlock: toBlock}
-  ));
-  console.log('Getting events info for token')
-  DXdaoSnapshot.token.events = DXdaoSnapshot.token.events.concat( await dxToken.getPastEvents(
-    'allEvents', {fromBlock: fromBlock, toBlock: toBlock}
-  ));
-  console.log('Getting events info for reputation')
-  DXdaoSnapshot.reputation.events = DXdaoSnapshot.reputation.events.concat( await dxReputation.getPastEvents(
-    'allEvents', {fromBlock: fromBlock, toBlock: toBlock}
-  ));
-  console.log('Getting events info for genesisProtocol')
-  DXdaoSnapshot.genesisProtocol.events = DXdaoSnapshot.genesisProtocol.events.concat( await genesisProtocol.getPastEvents(
-    'allEvents', {fromBlock: fromBlock, toBlock: toBlock}
-  ));
-  
-  console.log('Getting txs from schemes..')
-  for (var schemeAddress in schemes) {
-    console.log('Getting txs from scheme', schemeAddress);
-    await sleep(30000);
-    if (schemes.hasOwnProperty(schemeAddress)) {
-      transactionsFetched = await getTransactions(schemeAddress, fromBlock, toBlock);
-      if (!DXdaoSnapshot.schemes[schemeAddress])
-        DXdaoSnapshot.schemes[schemeAddress] = { txs: [], internalTxs: [], events: [] };
-      DXdaoSnapshot.schemes[schemeAddress].txs = DXdaoSnapshot.token.txs.concat(transactionsFetched.txs)
-      DXdaoSnapshot.schemes[schemeAddress].internalTxs = DXdaoSnapshot.token.internalTxs.concat(transactionsFetched.internalTxs)
-      DXdaoSnapshot.schemes[schemeAddress].events = DXdaoSnapshot.schemes[schemeAddress].events.concat(
-        await schemes[schemeAddress].getPastEvents(
-        'allEvents', {fromBlock: fromBlock, toBlock: toBlock}
-        )
-      );
-      _.remove(DXdaoSnapshot.schemes[schemeAddress].events, (contractEvent) => {
-        return contractEvent.returnValues._avatar && (contractEvent.returnValues._avatar != dxAvatar.address)
-      });
-    }
-  }
-  
-  fs.writeFileSync('DXdaoSnapshot.json', JSON.stringify(DXdaoSnapshot, null, 2), {encoding:'utf8',flag:'w'});
-} 
-
-Promise.all([main()]).then(process.exit);
+const fs = require('fs');

```

Comment:
Weird that I see the whole file rewritten..is difficult to review it that way.

Reply:
I'm not sure why this is... very strange.